provider/alibabacloud/alibaba_cloud.go

```diff
@@ -373,24 +374,23 @@ func (p *AlibabaCloudProvider) groupRecords(records []alidns.Record) (endpointMa
 func (p *AlibabaCloudProvider) records() ([]alidns.Record, error) {
 	log.Infof("Retrieving Alibaba Cloud DNS Domain Records")
 	var results []alidns.Record
-
-	if len(p.domainFilter.Filters) == 0 {
-		domainNames, tmpErr := p.getDomainList()
-		if tmpErr != nil {
-			log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
-			return results, tmpErr
-		}
-		for _, tmpDomainName := range domainNames {
-			tmpResults, err := p.getDomainRecords(tmpDomainName)
+	hostedZoneDomains, err := p.getDomainList()
+	if err != nil {
+		log.Errorf("getDomainList: %v", err)
+		return results, err
```

Comment:
Don't both log and return an error; that will lead to double-logging.

```suggestion
		return results, fmt.Errorf("getting domain list: %w", err)
```